### PR TITLE
fix(build_deploy): Only Create SC Tag When Building SC Branch

### PR DESCRIPTION
**Pull request by casey-williams-rh**
## Overview
It has been observed  that the `build_deploy` script is set up so that when the SC build Job runs, it write a new image over the original `image_tag` on which the commit is based. 

This update fixes that issue. 

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: build_deploy.sh [hunk 1]
```diff
@@ -7,16 +7,15 @@ CICD_TOOLS_URL="https://raw.githubusercontent.com/RedHatInsights/cicd-tools/main
 source <(curl -sSL "$CICD_TOOLS_URL") image_builder
 
 export CICD_IMAGE_BUILDER_IMAGE_NAME='quay.io/cloudservices/compliance-backend'
-export CICD_IMAGE_BUILDER_BUILD_ARGS=("IMAGE_TAG=$(cicd::image_builder::get_image_tag)")
+
```
Comment (skateman) [inline build_deploy.sh:10]: This change would also affect the non-sc branch 😞 did you check that if it would not cause any side-effects?

Comment (casey-williams-rh) [reply build_deploy.sh:10]:
```
export CICD_IMAGE_BUILDER_BUILD_ARGS=("IMAGE_TAG=$(cicd::image_builder::get_image_tag)")
````
This line is in the `else` statement. If the branch is not `"origin/security-compliance"` the execution will be the same as before.

**Comment by skateman** [comment]
@casey-williams-rh I think the commit message should not start with a capital letter after the `:`, that's why the static analysis fails

**Comment by casey-williams-rh** [comment]
@skateman , commit message has been updated.

**Review by skateman — APPROVED** [review]
The Seal of Approval

**Pull request merged**